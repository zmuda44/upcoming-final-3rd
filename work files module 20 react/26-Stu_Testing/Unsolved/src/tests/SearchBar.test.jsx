import pretty from 'pretty';
import { render } from '@testing-library/react';
import SearchBar from '../components/SearchBar';

describe('SearchBar', () => {
  // In this example, we render the SearchBar component and format the rendered HTML using the pretty package before saving it as a snapshot
  it('should render and match snapshot', () => {
    act(() => {
      // Render the component
      render(<SearchBar />, container);
    });

    // Format the rendered output
    const html = pretty(container.innerHTML);

    // Verify the output
    expect(html).toMatchSnapshot();
  });
});
