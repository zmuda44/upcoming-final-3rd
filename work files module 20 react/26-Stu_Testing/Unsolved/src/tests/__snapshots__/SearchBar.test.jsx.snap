// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchBar > should render and match snapshot 1`] = `
"<div class=\\"search-bar ui segment\\">
  <form class=\\"ui form\\">
    <div class=\\"field\\"><label>Retrieve GitHub Issues</label><input type=\\"text\\" placeholder=\\"facebook/react\\" value=\\"microsoft/vscode\\"></div>
  </form>
</div>"
`;

exports[`SearchBar should render and match snapshot 1`] = `
"<div class=\\"search-bar ui segment\\">
  <form class=\\"ui form\\">
    <div class=\\"field\\"><label>Retrieve GitHub Issues</label><input type=\\"text\\" placeholder=\\"facebook/react\\" value=\\"microsoft/vscode\\"></div>
  </form>
</div>"
`;
